Plotting and metrics validation

mlgill opened the pull request:
Plotting additions to the code

mlgill committed:
Plotting

mlgill committed:
Merge branch 'rilango/bm_refactor' into mgill/bm_plotting

mlgill committed:
Create CDDD dataset also

mlgill committed:
Merge branch 'mgill/bm_plotting' of github.com:mlgill/cheminformatics into mgill/bm_plotting

mlgill committed:
Remove plotting for now

mlgill committed:
Update replace SQL

mlgill committed:
Cleanup

mlgill committed:
Fix path

mlgill committed:
Add test

Commit:
Merge branch 'rilango/bm_refactor' into mgill/bm_plotting

mlgill committed:
BUG FIX: ensure index not included in fingerprints features

mlgill committed:
BUG FIX: set data input size appropriately for dataset

mlgill committed:
BUG FIX: fix inconsistent indexes

mlgill committed:
Ensure CDDD class is importable

mlgill committed:
BUG FIX: ensure valid sampled molecules are canonicalized

mlgill committed:
Accidentally committed files

mlgill committed:
Move UI clustering benchmark

mlgill committed:
Move UI clustering benchmark

mlgill committed:
Move benchmark data prep scripts

mlgill committed:
Training data added to SQL after each file

mlgill committed:
Move yaml file to config dir

Commit:
Minor bugfixes

mlgill committed:
Add logging to train debug generatio

mlgill committed:
Merge branch 'mgill/bm_plotting' into mgill/bm_plotting

mlgill committed:
Merge pull request #1 from rilango/mgill/bm_plotting

Difference between two refactoring related commit.

mlgill committed:
Revert one change from merge

mlgill committed:
Move config file

mlgill committed:
Improve error checking for CSV data

mlgill committed:
Sampling metrics validated

mlgill committed:
Changes to return best predictions for plotting

mlgill committed:
Bug fix: bioactivity fingerprints & dataset loading

rilango commented on benchmark/cuchembm/__main__.py:
```diff
@@ -48,16 +58,43 @@ def wait_for_megamolbart_service(inferrer):
+    return input_size
+
+
+@hydra.main(config_path="config", config_name="benchmark_metrics")
```
This will limit us to use only one configuration file, which also means we will lose the flexibility to run more then one instance of bm run on one machine

mlgill replied:
Have updated. Thanks.

mlgill committed:
Fix ML model params

mlgill committed:
Fix config path in __main__.py

mlgill committed:
Update benchmark metrics yaml

mlgill committed:
Add seaborn to container

rilango committed:
Changes to accept a inference wrapper outside of codebase.

rilango committed:
Merge remote-tracking branch 'mgill/mgill/bm_plotting' into mgill/bm_plotting

rilango committed:
Further optimization of benchmark code...

- Separate out SMILES generation code.
  The idea is not to generate SMILES during benchmark testing. Almost
  always, we are generating the SMILES prior to running the benchmark
  code. Therefore, separing out this local generated SMILES db will
  make it easier to automate these steps and keep the code clean.

rilango committed:
Add methods to fetch from local db containing generated molecules.

Commit:
Notebook to find the intersection dataset between CDDD training data and
benchmarking dataset.

Commit:
Merge branch 'mgill/bm_plotting' of github.com:mlgill/cheminformatics into mgill/bm_plotting

Commit:
Include the dataset size for context.

mlgill committed:
Remove CDDD train from ZINC15 test split

mlgill committed:
Merge branch 'mgill/bm_plotting' of github.com:mlgill/cheminformatics into mgill/bm_plotting

Commit:
Update to list the common SMILES

Commit:
Merge branch 'mgill/bm_plotting' of github.com:mlgill/cheminformatics into mgill/bm_plotting

rilango committed:
Changes to create cache with additional details to reduce the time
required to run benchmark runs.

rilango committed:
Merge remote-tracking branch 'mgill/mgill/bm_plotting' into rilango/cleanup

mlgill committed:
Normalize data

mlgill committed:
Merge branch 'mgill/bm_plotting' of github.com:mlgill/cheminformatics into mgill/bm_plotting

mlgill committed:
Plotting completed

mlgill committed:
Nearest neighbor fix

mlgill committed:
Configurable normalization of embeddings

rilango committed:
Add the code to compute Validity, Uniqueness and Novelity ratio

rilango committed:
Change to include SMILES from all datasets.

rilango committed:
Merge remote-tracking branch 'mgill/mgill/bm_plotting' into rilango/cleanup

rilango committed:
Changes to improve the queries to allow multiple parameter

Commit:
Change to introduce parallel query for generating sample

Commit:
Fix bugs

Commit:
Cleanup

rilango committed:
Minor bugs

Commit:
Bug fix and adjustments for parallelization

rilango committed:
Minor bug

rilango committed:
Changes made for embedding to work

rilango committed:
Fix a mistake in commit.

rilango committed:
Merge branch 'rilango/cleanup' of github.com:rilango/cheminformatics into rilango/cleanup

rilango committed:
Bugfix: Input_size for bioactivity was not use as intended.

rilango committed:
Merge branch 'rilango/cleanup' of github.com:rilango/cheminformatics into rilango/cleanup

Commit:
Changes to remote individual databases and start using a commond db for all metrics.

mlgill commented:
@rilango Just a note that this is still awaiting review. Let me know tomorrow if anything is needed to merge.

Commit:
Changes to improve runtime perf.

Commit:
Fix  review comments.

mlgill committed:
Merge pull request #2 from rilango/rilango/cleanup

Performance and cleanup

mlgill committed:
BUG FIX: ensure valid sampled molecules are canonicalized
